app/models/data_export.rb

```diff
@@ -26,14 +26,30 @@ class DataExport < ApplicationRecord
   include DivisionBased
 
   belongs_to :division
-  has_many :attachments, as: :media_attachable, dependent: :nullify
+  has_many :attachments, as: :media_attachable, dependent: :nullify, class_name: "Media"
 
   before_save :set_name
 
   DATA_EXPORT_TYPES = {
     "data_export" => "DataExport"
   }
 
+  def to_csv!
+    raise ArgumentError, "No data found" unless data.present?
+    raise TypeError, "Data invalid" unless (data.is_a?(Array) && data.first.is_a?(Array))
```

Comment:
Should the message specify that data should be a 2D array? or is it expected a developer will look here and see that from the code?

Reply:
that `data` needs to be a 2D array is part of the assumptions of this API, that should probably be documented somewhere, but in practice nobody should be running into this error because nobody should be writing code to add non-2d-array data to that field, so the message is not a huge deal